### DISCUSSION TITLE
how to run ICeWall test examples

### DISCUSSION THREAD

**Discussion by vijayank88** [The-OpenROAD-Project/OpenROAD#1164] [Q&A] [answered]
Hi,

I am trying to run OpenROAD/src/ICeWall/test examples. Used below command inside openroad and stopped with error

openroad 1> source soc_bsg_black_parrot_nangate45.tcl 
[WARNING ODB-0229] Error: library (NangateOpenCellLibrary.mod) already exists
[WARNING ODB-0229] Error: library (dummy_pads) already exists
[WARNING STA-0053] dummy_pads.lib line 7, library dummy_pads_worst_case already exists.
Segmentation fault (core dumped)

Anyone can explain how to run this flow and to view DEF file klayout to view my chip level io connection created by ICeWall

Thanks in advance

**Comment by maliberty** [comment]
@Colin-Holehouse @ibrahimkhairy can you help?

**Reply by vijayank88** [reply]
Any update on this query?

**Reply by Colin-Holehouse** [reply]
The error messages indicate that openraod is complaining about libraries being loaded multiple times. This would occur if you start openroad specifying a script on the command line, then source the script will give this error:

```
% openroad soc_bsg_black_parrot_nangate45.tcl
OpenROAD v2.0-993-g8a2c777a3
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
[INFO ODB-0222] Reading LEF file: NangateOpenCellLibrary.mod.lef
[INFO ODB-0223]     Created 22 technology layers
[INFO ODB-0224]     Created 27 technology vias
[INFO ODB-0225]     Created 134 library cells
[INFO ODB-0226] Finished LEF file:  NangateOpenCellLibrary.mod.lef
[INFO ODB-0222] Reading LEF file: dummy_pads.lef
[INFO ODB-0225]     Created 24 library cells
[INFO ODB-0226] Finished LEF file:  dummy_pads.lef
[INFO IFP-0001] Added 1885 rows of 13893 sites.
[INFO PAD-0053] Creating padring nets: RETN_10, RETN_11, RETN_12, RETN_13, RETN_14, RETN_15, RETN_16, RETN_17, RETN_18, RETN_19, RETN_20, RETN_0, RETN_1, RETN_2, RETN_3, RETN_4, RETN_5, RETN_6, RETN_7, RETN_8, RETN_9, SNS_10, SNS_11, SNS_12, SNS_13, SNS_14, SNS_15, SNS_16, SNS_0, SNS_17, SNS_1, SNS_18, SNS_2, SNS_19, SNS_20, SNS_3, SNS_4, SNS_5, SNS_6, SNS_7, SNS_8, SNS_9, DVSS_0, DVSS_1, DVDD_0, DVDD_1.
[INFO PAD-0051] Creating padring net: VSS.
[INFO PAD-0051] Creating padring net: VDD.
No differences found.
openroad> source soc_bsg_black_parrot_nangate45.tcl
[WARNING ODB-0229] Error: library (NangateOpenCellLibrary.mod) already exists
[WARNING ODB-0229] Error: library (dummy_pads) already exists
[WARNING STA-0053] dummy_pads.lib line 7, library dummy_pads_worst_case already exists.
Segmentation fault (core dumped)

```

If you start openroad with no script argument, then there will be no conflicts with multiply loaded libraries when you execute the source command.

**Reply by dralabeing** [reply]
This should still not cause a seg fault

**Reply by Colin-Holehouse** [reply]
@maliberty - can you address the issue of multiple loads of the same library causing a seg fault - this is outside the scope of ICeWall

**Comment by maliberty** [comment]
@jjcherry56 the crash is in 

sta::ConcreteInstance::deletePin 
sta::ConcreteNetwork::deletePin 
sta::ConcreteNetwork::deleteInstance 
sta::ConcreteNetwork::deleteTopInstance 
sta::ConcreteNetwork::linkNetwork 
ord::dbLinkDesign 

reproduce by going to src/ICeWall/test and running
% openroad soc_bsg_black_parrot_nangate45.tcl
[at OR prompt] source soc_bsg_black_parrot_nangate45.tcl

**Comment by vijayank88** [comment]
[vijayan@ctpl-hp11 test]$ ../../../build/src/openroad 
OpenROAD 1 3a8b060d9026de9b35e4f35ee59157ce9416c3ff
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
openroad 1> source soc_bsg_black_parrot_nangate45.tcl 
[INFO ODB-0222] Reading LEF file: NangateOpenCellLibrary.mod.lef
[INFO ODB-0223]     Created 22 technology layers
[INFO ODB-0224]     Created 27 technology vias
[INFO ODB-0225]     Created 134 library cells
[INFO ODB-0226] Finished LEF file:  NangateOpenCellLibrary.mod.lef
[INFO ODB-0222] Reading LEF file: dummy_pads.lef
[INFO ODB-0225]     Created 24 library cells
[INFO ODB-0226] Finished LEF file:  dummy_pads.lef
[INFO IFP-0001] Added 1885 rows of 13893 sites.
[INFO PAD-0053] Creating padring nets: RETN_10, RETN_11, RETN_12, RETN_13, RETN_14, RETN_15, RETN_16, RETN_17, RETN_18, RETN_19, RETN_20, RETN_0, RETN_1, RETN_2, RETN_3, RETN_4, RETN_5, RETN_6, RETN_7, RETN_8, RETN_9, SNS_10, SNS_11, SNS_12, SNS_13, SNS_14, SNS_15, SNS_16, SNS_0, SNS_17, SNS_1, SNS_18, SNS_2, SNS_19, SNS_20, SNS_3, SNS_4, SNS_5, SNS_6, SNS_7, SNS_8, SNS_9, DVSS_0, DVSS_1, DVDD_0, DVDD_1.
[INFO PAD-0051] Creating padring net: VSS.
[INFO PAD-0051] Creating padring net: VDD.
No differences found.

Is the above log right?

**Comment by maliberty** [comment, marked as answer]
"No differences found. " tells you it is right.

**Reply by vijayank88** [reply]
@Colin-Holehouse @maliberty  thanks. Its docker based run. Multiple run will cause seg fault issue.
Also with openroad -gui I can run source soc_bsg_black_parrot_nangate45.tcl able to see its graphical representation.

**Reply by dralabeing** [reply]
@vijayank88 - What you are saying is that for docker based runs,  there needs be a cleanup option as well. Please elaborate what you found here.
